api/nsp/v1/diff.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright (c) 2021-2022 Nordix Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "fmt"
+
+// Checks the differences between 2 port nat list
+// 1st return gives the items added
+// 2nd return gives the items in common
+// 3rd return gives the items removed
+func PortNatDiff(old []*Conduit_PortNat, new []*Conduit_PortNat) ([]*Conduit_PortNat, []*Conduit_PortNat, []*Conduit_PortNat) {
```

Comment:
This is not a set-diff as the name implies, a more appropriate name would be something like "TwoWayDiffAndIntersection", which is _not_ a good function in an API. Rewrite to a proper diff, e.g.
```go
func PortNatDiff(set1 []*Conduit_PortNat, set1 []*Conduit_PortNat) []*Conduit_PortNat
```
The über-complex function is not needed (more about that later)